pennylane/math/multi_dispatch.py

```diff
@@ -340,3 +374,50 @@ def where(condition, x, y):
     tensor([ 0.6000,  0.2300,  0.7000, -4.0000, -5.0000], grad_fn=<SWhereBackward>)
     """
     return np.where(condition, x, y, like=_multi_dispatch([condition, x, y]))
+
+
+def unwrap(values, max_depth=None):
+    """Unwrap a sequence of objects to NumPy arrays.
+
+    Note that tensors on GPUs will automatically be copied
+    to the CPU.
+
+    Args:
+        values (Sequence[tensor_like]): sequence of tensor-like objects to unwrap
+        max_depth (int): Positive integer indicating the depth of unwrapping to perform
+            for nested tensor-objects. This argument only applies when unwrapping
+            Autograd ``ArrayBox`` objects.
+
+    **Example**
+
+    >>> values = [np.array([0.1, 0.2]), torch.tensor(0.1, dtype=torch.float64), torch.tensor([0.5, 0.2])]
+    >>> math.unwrap(values)
+    [array([0.1, 0.2]), 0.1, array([0.5, 0.2], dtype=float32)]
+
+    This function will continue to work during backpropagation:
+
+    >>> def cost_fn(params):
+    ...     unwrapped_params = math.unwrap(params)
+    ...     print("Unwrapped:", [(i, type(i)) for i in unwrapped_params])
+    ...     return np.sum(np.sin(params))
+    >>> params = np.array([0.1, 0.2, 0.3])
+    >>> grad = autograd.grad(cost_fn)(params)
+    Unwrapped: [(0.1, <class 'float'>), (0.2, <class 'float'>), (0.3, <class 'float'>)]
+    >>> print(grad)
+    [0.99500417 0.98006658 0.95533649]
+    """
+    res = []
+
+    for t in values:
+        if isinstance(t, ArrayBox):
+            a = np.to_numpy(t, max_depth=max_depth)
+        else:
+            a = np.to_numpy(t)
+
+        if isinstance(a, ndarray) and not a.shape:
+            # if NumPy array is scalar, convert to a Python float

```

Comment:
what if we have something like `tf.Variable(0.1)` ? Would you also expect a float? In other words, do we always "squeeze" scalars here?

Reply:
We do. This is an issue we have with some plugins.

Even though `default.qubit` treats `np.array(0.1)` the same as `0.1`, not all QC frameworks do. In particular, Qiskit and Braket have logic that will fail if given a scalar array!

c.f. https://github.com/PennyLaneAI/pennylane/pull/893